packages/authentication/src/__tests__/acceptance/basic-auth.acceptance.ts

```diff
@@ -87,10 +85,45 @@ describe.skip('Basic Authentication', () => {
     });
   }
 
+  // Since it has to be user's job to provide the `verify` function and
+  // instantiate the passport strategy, we cannot add the imported `BasicStrategy`
+  // class as extension directly, we need to wrap it as a strategy provider,
+  // then add the provider class as the extension.
+  // See Line 89 in the function `givenAServer`
+  class PassportBasicAuthProvider implements Provider<AuthenticationStrategy> {
+    value(): AuthenticationStrategy {
+      const basicStrategy = this.configuratedBasicStrategy(verify);
+      return this.convertToAuthStrategy(basicStrategy);
+    }
+
+    configuratedBasicStrategy(verifyFn: BasicVerifyFunction): BasicStrategy {
+      return new BasicStrategy(verifyFn);
+    }
+
+    convertToAuthStrategy(basic: BasicStrategy): AuthenticationStrategy {
+      return new StrategyAdapter(basic, 'basic');
```

Comment:
@jannyHou , let's store the 'basic' identifier string for the authentication strategy in a constant somewhere else.